src/modules/fw_att_control/FixedwingAttitudeControl.cpp

```diff
@@ -466,30 +468,31 @@ FixedwingAttitudeControl::vehicle_land_detected_poll()
 	}
 }
 
-void FixedwingAttitudeControl::get_airspeed_and_scaling(float &airspeed, float &airspeed_scaling)
+float FixedwingAttitudeControl::get_airspeed_and_update_scaling()
 {
+	_airspeed_sub.update();
 	const bool airspeed_valid = PX4_ISFINITE(_airspeed_sub.get().indicated_airspeed_m_s)
-				    && (_airspeed_sub.get().indicated_airspeed_m_s > 0.0f)
-				    && (hrt_elapsed_time(&_airspeed_sub.get().timestamp) < 1e6);
+				    && (_airspeed_sub.get().indicated_airspeed_m_s >= 0.0f)
+				    && (hrt_elapsed_time(&_airspeed_sub.get().timestamp) < 1_s);
+
+	if (!airspeed_valid) {
+		perf_count(_nonfinite_input_perf);
+	}
+
+	// if no airspeed measurement is available out best guess is to use the trim airspeed
+	float airspeed = _parameters.airspeed_trim;
 
 	if (!_parameters.airspeed_disabled && airspeed_valid) {
 		/* prevent numerical drama by requiring 0.5 m/s minimal speed */
 		airspeed = math::max(0.5f, _airspeed_sub.get().indicated_airspeed_m_s);
```

Comment:
We force a min of 0.5 here - this could be turned into a configurable parameter. The main reason to use a parameter instead is that it would allow to tune the scaling to a more reasonable number, as you get pretty much full deflection at 0.5 m/s with this scaling.

Reply:
@LorenzMeier No, the airspeed used for the scaling is always constrained to a minimum of "airspeed_min".

Reply:
Ok - but does that mean we have too little scaling during runway takeoffs and alike situations where we are way below min?

Reply:
I agree with Lorenz. We have examples of hand launch takeoff with our fixed wing where the airspeed is low and the system is not trying hard enough to react, presumably due to not enough gain scaling.

Reply:
Can you file an issue with an example log?

Reply:
@Antiheavy @LorenzMeier Personally I'm not too  happy that we are using the min airspeed parameter when we actually mean "stall airspeed". When you fly ALT mode and you have the stick full down then the system will go to min airspeed and that is hopefully above the stall speed.
Paul introduced a stall airspeed in his PR for the airspeed fault detection so maybe we can make use of that.

Reply:
@RomanBapst if you have time we should talk about https://github.com/PX4/Firmware/pull/11846.
One of the minor things I wanted to fix before merging was adding the stall airspeed to the fixedwing controllers rather than commander.

Reply:
@dagar Ok, basically from tomorrow on I have bandwidth to fully concentrate on that one.
@sfuhrer is taking care of the last pieces for VTOL (basically the tail-sitter problem)

Reply:
> Can you file an issue with an example log?

@LorenzMeier Yes, I will find some logs and create a separate issue. thanks.

Reply:
> Personally I'm not too happy that we are using the min airspeed parameter when we actually mean "stall airspeed".

Yes I agree completely!  Ideall we should be using a "stall speed" param for this and several other things (e.g. the airspeed failsafe).